packages/jobs/src/queues/index.ts

```diff
@@ -17,37 +23,75 @@ type JobEnqueueFn = {
   ) => Promise<Job<JobDefinition[Queues][Jobs]['data']>>
 }
 
+const attempts = process.env.NODE_ENV === 'production' ? 100 : 3
+
+export const DEFAULT_JOB_OPTIONS: JobsOptions = {
+  attempts,
+  backoff: {
+    type: 'exponential',
+    delay: 1000,
+  },
+}
+
 function setupQueue({
   name,
   connection,
   jobs,
 }: {
   name: Queues
   connection: Redis
-  jobs: Jobs[]
+  jobs: readonly QueueJob[]
 }) {
-  const queue = new Queue(name, { connection })
+  const queue = new Queue(name, {
+    connection,
+    defaultJobOptions: DEFAULT_JOB_OPTIONS,
+  })
+  const jobz = jobs.reduce((acc, job) => {
+    const key = `enqueue${capitalize(job.name)}` as EnqueueFunctionName<
+      typeof job.name
+    >
+    const enqueueFn = (
+      params: JobDefinition[typeof name][Jobs]['data'],
+      options: JobsOptions,
+    ) => queue.add(job.name, params, options)
+
+    return { ...acc, [key]: enqueueFn }
+  }, {} as JobEnqueueFn)
+
   return {
+    queue,
     events: new QueueEvents(name, { connection }),
-    jobs: jobs.reduce((acc, jobName) => {
-      const key = `enqueue${capitalize(jobName)}`
-      const enqueuFn = (
-        params: JobDefinition[typeof name][typeof jobName]['data'],
-        options?: JobsOptions,
-      ) => {
-        queue.add(jobName, params, options)
-      }
-      return { ...acc, [key]: enqueuFn }
-    }, {} as JobEnqueueFn),
+    jobs: jobz,
   }
 }
 
+export const QUEUES = {
+  [Queues.defaultQueue]: {
+    name: Queues.defaultQueue,
+    jobs: [createProviderLogJob, createDocumentLogJob],
+  },
+  [Queues.eventsQueue]: {
+    name: Queues.eventsQueue,
+    jobs: [publishEventJob],
+  },
+  [Queues.eventHandlersQueue]: {
+    name: Queues.eventHandlersQueue,
+    jobs: Object.values(EventHandlers).flat(),
+  },
+} as const
+
+type QueueJob = (typeof QUEUES)[keyof typeof QUEUES]['jobs'][number]
+
 export function setupQueues({ connection }: { connection: Redis }) {
-  return {
-    [Queues.defaultQueue]: setupQueue({
-      connection,
-      name: Queues.defaultQueue,
-      jobs: [Jobs.createProviderLogJob, Jobs.createDocumentLogJob],
-    }),
-  }
+  return Object.entries(QUEUES).reduce<{
+    [K in keyof typeof QUEUES]: ReturnType<typeof setupQueue>
+  }>(
+    (acc, [name, { jobs }]) => {
+      return {
+        ...acc,
+        [name]: setupQueue({ name: name as Queues, connection, jobs }),
+      }
+    },
+    {} as { [K in keyof typeof QUEUES]: ReturnType<typeof setupQueue> },
+  )
```

Comment:
lots of typescript crazyness just to make the type declarations generalize to more than one queue (previously it only worked with default queue)